cli/flox/src/commands/init/python.rs

```diff
@@ -902,46 +901,28 @@ mod tests {
     async fn pyproject_empty_with_catalog() {
         let (mut flox, _temp_dir_handle) = flox_instance();
 
-        if let Client::Mock(ref mut client) = flox.catalog_client {
-            // Response for unconstrained python version
-            client.push_resolve_response(vec![resolved_pkg_group_with_dummy_package(
-                "python3_group",
-                &System::from("aarch64-darwin"),
-                "python3",
-                "python3",
-                "3.11.6",
-            )]);
-        }
+        flox.catalog_client = auto_recording_catalog_client("python_no_pyproject");
```

Comment:
suggestion nonblocking: unless we're sharing this with other tests, I think I have a preference for just using the unit test name for the mock name, so it's not confusing what file belongs to which test

Reply:
I did briefly look at whether we could name them automatically but it seems non-trivial based on [this](https://internals.rust-lang.org/t/discovering-the-current-test-name/15325).

I suspect we're going to forget to rename them when making changes either way and whilst arguably we might change scenarios more often than we rename tests, having changed both in this PR alone and having not noticed the convention when I reviewed https://github.com/flox/flox/pull/3077, I think we're more likely to remember to rename scenarios and it matches the existing conventions in `mk-data`.